openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -643,6 +672,87 @@ def get_allowed_block_types(library_key):  # pylint: disable=unused-argument
     return info
 
 
+def get_bundle_links(library_key):
+    """
+    Get the list of bundles/libraries linked to this content library.
+
+    Returns LibraryBundleLink objects (defined above).
+
+    Because every content library is a blockstore bundle, it can have "links" to
+    other bundles, which may or may not be content libraries. This allows using
+    XBlocks (or perhaps even static assets etc.) from another bundle without
+    needing to duplicate/copy the data.
+
+    Links always point to a specific published version of the target bundle.
+    Links are identified by a slug-like ID, e.g. "link1"
+    """
+    ref = ContentLibrary.objects.get_by_key(library_key)
+    links = blockstore_cache.get_bundle_draft_direct_links_cached(ref.bundle_uuid, DRAFT_NAME)
+    results = []
+    for link_name, link_data in links.items():
+        # Is this linked bundle a content library?
+        try:
+            opaque_key = ContentLibrary.objects.get(bundle_uuid=link_data.bundle_uuid).library_key
+        except ContentLibrary.DoesNotExist:
+            opaque_key = None
```

Comment:
Do we need to do this check? I ask mostly because it seems like it could get expensive if the number of links starts to become larger.

Reply:
> Do we need to do this check? I ask mostly because it seems like it could get expensive if the number of links starts to become larger.

For now it's not really important as all bundles are libraries. In the future it would matter more.

I can either replace this check with a comment "# TODO: If we support other types of bundles, check if this is a library and if not set the key to None", or I can refactor this to pull the MySQL `.get()` out of the loop and make it a single query for all linked bundles. Any preference?

Reply:
Refactored in 6b5730a8fa3bb6d0bac4372290ec9ca02e895385 so this makes only a single database query for any number of links. BTW, it's not just a check; it's necessary to query the DB to get the `org` ID and `slug` in order to generate the opaque key string from just the bundle UUID.